src/test/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStepTest.java

```diff
@@ -186,16 +197,26 @@ public InvalidChangelogSCM() { }
         @Override public void checkout(Run<?,?> build, Launcher launcher, FilePath workspace, TaskListener listener, File changelogFile, SCMRevisionState baseline) throws IOException, InterruptedException {
             // Unlike the superclass, which adds an XML header to the file, we just ignore the file.
         }
-        @Override public ChangeLogParser createChangeLogParser() {
-            return new ChangeLogParser() {
-                public ChangeLogSet<? extends ChangeLogSet.Entry> parse(Run build, RepositoryBrowser<?> browser, File changelogFile) throws IOException, SAXException {
-                    Jenkins.XSTREAM2.fromXML(changelogFile);
-                    throw new AssertionError("The previous line should always fail because the file is empty");
-                }
-            };
-        }
```

Comment:
To allow the new unit test to work, I copied this static class, thus creating the new `public static class FakeChangeLogSCM extends NullSCM`. Because this `@Override public ChangeLogParser` had to be used by both, I separated that out into its own static method on L213-220.